### PR TITLE
Feature/update pet

### PR DESCRIPTION
**Pull request by andersonnaz**
Create breed and size Entity models
Create Breed and Size Repository
Create AppointPet use case to compile all use cases(specieAppoint, breedAppoint, sizeAppoint) at the same place.
Create validators to breed and size
Update Addpet including breed and size
Update breed seed with some new breeds
Update size seed with some new sizes
Update doc

### CODE REVIEW

File: tests/src/data/use-cases/pet/db-add-pet.spec.ts [hunk 1]
```diff
@@ -78,25 +84,25 @@ describe('DbAddPet Use Case', () => {
...
 
-      const promise = sut.add(params)
+  //     const promise = sut.add(params)
 
-      await expect(promise).rejects.toThrow()
-    })
-  })
+  //     await expect(promise).rejects.toThrow()
+  //   })
+  // })
```
Comment (matheusgondra) [inline tests/src/data/use-cases/pet/db-add-pet.spec.ts]: Remove this commented test

File: tests/src/data/use-cases/pet/db-appoint-breed.spec.ts
Comment (matheusgondra) [inline tests/src/data/use-cases/pet/db-appoint-breed.spec.ts:1]: Replace the `test` function in jest with the `it` function, as this is the one we are using by convention

Comment (matheusgondra) [reply tests/src/data/use-cases/pet/db-appoint-breed.spec.ts:1]: Put the test description in capital letters at the beginning of the sentence

File: tests/src/data/use-cases/pet/db-appoint-pet.spec.ts
Comment (matheusgondra) [inline tests/src/data/use-cases/pet/db-appoint-pet.spec.ts:1]: Replace the test function in jest with the it function, as this is the one we are using by convention. Put the test description in capital letters at the beginning of the sentence

File: tests/src/data/use-cases/pet/db-appoint-pet.spec.ts [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        name: 'Outros'
+      }
+      const anyBreed = {
+        id: 'any_id',
+        name: 'any_name'
+      }
+      const anySize = {
+        id: 'any_id',
+        name: 'any_name'
+      }
```
Comment (matheusgondra) [inline tests/src/data/use-cases/pet/db-appoint-pet.spec.ts]: This piece of code could be moved to a separate function to remove code repetition in other tests

File: tests/src/application/validation/validators/size-validation.spec.ts
Comment (matheusgondra) [inline tests/src/application/validation/validators/size-validation.spec.ts:1]: Replace the test function in jest with the it function, as this is the one we are using by convention. Put the test description in capital letters at the beginning of the sentence

File: tests/src/application/validation/validators/breed-validation.spec.ts
Comment (matheusgondra) [inline tests/src/application/validation/validators/breed-validation.spec.ts:1]: Replace the test function in jest with the it function, as this is the one we are using by convention. Put the test description in capital letters at the beginning of the sentence

**Review by matheusgondra — CHANGES_REQUESTED** [review]

File: src/application/validation/validators/breed-validation.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { type Validation } from '@/application/protocols'
+import { InvalidParamError } from '@/application/errors'
+import { type BreedValidator } from '../protocols/breed-validator'
```
Comment (rodrigofmeneses) [inline src/application/validation/validators/breed-validation.ts]: follow "@/app..." pattern

File: src/application/validation/validators/size-validation.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { type Validation } from '@/application/protocols'
+import { type SizeValidator } from '../protocols'
```
Comment (rodrigofmeneses) [inline src/application/validation/validators/size-validation.ts]: here too
https://github.com/PetJournal/petjournal.api/pull/48/files#r1452743285

File: src/data/use-cases/pet/db-appoint-pet.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
+import { type LoadBreedByNameRepository, type LoadSpecieByNameRepository } from '@/data/protocols'
+import { type Breed } from '@/domain/models/breed'
+import { type Size } from '@/domain/models/size'
+import { type Specie } from '@/domain/models/specie'
+import { type AppointPet } from '@/domain/use-cases'
+
+export class DbAppointPet implements AppointPet {
+  private readonly specieRepository: LoadSpecieByNameRepository
+  private readonly breedRepository: LoadBreedByNameRepository
+  private readonly sizeRepository: LoadSpecieByNameRepository
```
Comment (rodrigofmeneses) [inline src/data/use-cases/pet/db-appoint-pet.ts]: `LoadSpecieByNameRepository` is duplicated

File: tests/src/data/use-cases/pet/db-appoint-breed.spec.ts
Comment (andersonnaz) [reply tests/src/data/use-cases/pet/db-appoint-breed.spec.ts:1]: actually, this file is not be using, it's better to the code, remove it.

**Review by rodrigofmeneses — DISMISSED** [review]
LGFM

**Review by matheusgondra — APPROVED** [review]

**Review by rodrigofmeneses — APPROVED** [review]
LGFM

**Pull request merged**